### ISSUE TITLE
Remove asserts from the code

### ISSUE THREAD

**Issue by chessbr** [shuup/shuup#1417]
We shouldn't use assert to validate attributes and check logic. The reason is: https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement

> The current code generator emits no code for an assert statement when optimization is requested at compile time

So, whoever use this flag will miss these important validations, like did here https://github.com/shuup/shuup/blob/master/shuup/core/pricing/_context.py#L52

**Comment by tulimaki** [comment]
@suutari-ai Do you remember what we have talked about these in the past? I think these comes from the older iterations of Shuup and we have had these asserts here and there for a while.

**Comment by suutari-ai** [comment]
TL;DR: Any raised assertion error always means a programming error somewhere.

Asserts should never be used to validate bad input. They are a tool that just make sure that the code is being used as the developer intended and/or it does what the developer expects it to do.

As a guide line:
 * Bad user input should never trigger an assertion error. (E.g. query parameters, API calls, or whatever happens without writing new Python code)
 * Bad developer practices could trigger an assertion. I.e. if someone is calling a function with an integer argument if it was meant to be called with a list of strings.

Assertions are good in the code, because when running tests they check that the logic of the developer holds true. (E.g. I did this complex calculation here, and I think it should make this list `a` to always have all the elements of `b` in it, I'll add an assert to make sure that it holds: `assert all(x in a for x in b)`.

If the input to a function is from user, then proper way to check errors is with `if errorneous_condition...: raise ProperError('With a good message...')`.
